presto-geospatial/src/main/java/com/facebook/presto/geospatial/GeoFunctions.java

```diff
@@ -0,0 +1,493 @@
+import static com.facebook.presto.geospatial.GeometryUtils.POINT;
+import static com.facebook.presto.geospatial.GeometryUtils.POLYGON;
+import static com.facebook.presto.geospatial.GeometryUtils.deserialize;
+import static com.facebook.presto.geospatial.GeometryUtils.serialize;
+import static com.google.common.base.Preconditions.checkArgument;
+import static io.airlift.slice.Slices.EMPTY_SLICE;
+import static java.util.Arrays.asList;
+import static java.util.Objects.requireNonNull;
+
+public final class GeoFunctions
+{
+    private GeoFunctions() {}
+
+    @Description("Returns a Geometry type LineString object from Well-Known Text representation (WKT)")
+    @ScalarFunction("ST_LineFromText")
+    @SqlType(NAME)
+    public static Slice stLineString(@SqlType(StandardTypes.VARCHAR) Slice input)
+    {
+        OGCGeometry geometry = getGeometry(OGCGeometry.fromText(input.toStringUtf8()), "ST_LineFromText", LINE_STRING);
+        return serialize(geometry);
+    }
+
+    @Description("Returns a Geometry type Point object with the given coordinate values")
+    @ScalarFunction("ST_Point")
+    @SqlType(NAME)
+    public static Slice stPoint(@SqlType(StandardTypes.DOUBLE) double x, @SqlType(StandardTypes.DOUBLE) double y)
+    {
+        OGCGeometry geometry = OGCGeometry.createFromEsriGeometry(new Point(x, y), null);
+        return serialize(geometry);
+    }
+
+    @Description("Returns a Geometry type Polygon object from Well-Known Text representation (WKT)")
+    @ScalarFunction("ST_Polygon")
+    @SqlType(NAME)
+    public static Slice stPolygon(@SqlType(StandardTypes.VARCHAR) Slice input)
+    {
+        OGCGeometry geometry = getGeometry(OGCGeometry.fromText(input.toStringUtf8()), "ST_Polygon", POLYGON);
+        return serialize(geometry);
+    }
+
+    @Description("Returns the area of a polygon using Euclidean measurement on a 2D plane (based on spatial ref) in projected units")
+    @ScalarFunction("ST_Area")
+    @SqlType(StandardTypes.DOUBLE)
+    public static double stArea(@SqlType(NAME) Slice input)
+    {
+        OGCGeometry geometry = getGeometry(deserialize(input), "ST_Area", POLYGON);
+        return geometry.getEsriGeometry().calculateArea2D();
+    }
+
+    @Description("Returns a Geometry type object from Well-Known Text representation (WKT)")
+    @ScalarFunction("ST_GeometryFromText")
+    @SqlType(NAME)
+    public static Slice stGeometryFromText(@SqlType(StandardTypes.VARCHAR) Slice input)
+    {
+        return serialize(OGCGeometry.fromText(input.toStringUtf8()));
+    }
+
+    @SqlNullable
+    @Description("Returns the Well-Known Text (WKT) representation of the geometry")
+    @ScalarFunction("ST_AsText")
+    @SqlType(StandardTypes.VARCHAR)
+    public static Slice stAsText(@SqlType(NAME) Slice input)
+    {
+        return Slices.utf8Slice(deserialize(input).asText());
+    }
+
+    @Description("Returns the Point value that is the mathematical centroid of the set of points in Polygon")
```

Comment:
SQL/MM defines ST_Centroid as "returns the ST_Point value that is the mathematical centroid of the ST_Surface value" which is different from the implementation of this function. PostGIS implements ST_Centroid according to the standard [1].

Why do you want a different implementation for this function?

https://postgis.net/docs/ST_Centroid.html